src/sql/workbench/contrib/notebook/browser/notebookViews/notebookViews.component.ts

```diff
@@ -0,0 +1,283 @@
+		@Inject(IKeybindingService) private keybindingService: IKeybindingService,
+		@Inject(IContextMenuService) private contextMenuService: IContextMenuService,
+		@Inject(INotificationService) private notificationService: INotificationService,
+		@Inject(INotebookService) private notebookService: INotebookService,
+		@Inject(IConnectionManagementService) private connectionManagementService: IConnectionManagementService,
+		@Inject(IConfigurationService) private _configurationService: IConfigurationService,
+		@Inject(IEditorService) private _editorService: IEditorService
+	) {
+		super();
+		this._register(this._configurationService.onDidChangeConfiguration(e => {
+			this.previewFeaturesEnabled = this._configurationService.getValue('workbench.enablePreviewFeatures');
+		}));
+	}
+
+	public get notebookParams(): INotebookParams {
+		return this._notebookParams;
+	}
+
+	public get id(): string {
+		return this.notebookParams.notebookUri.toString();
+	}
+
+	isDirty(): boolean {
+		return this.notebookParams.input.isDirty();
+	}
+	isActive(): boolean {
+		return this._editorService.activeEditor ? this._editorService.activeEditor.matches(this.notebookParams.input) : false;
+	}
+	isVisible(): boolean {
+		let notebookEditor = this.notebookParams.input;
+		return this._editorService.visibleEditors.some(e => e.matches(notebookEditor));
+	}
+	executeEdits(edits: ISingleNotebookEditOperation[]): boolean {
+		throw new Error('Method not implemented.');
+	}
+	async runCell(cell: ICellModel): Promise<boolean> {
+		await this.modelReady;
+		let uriString = cell.cellUri.toString();
+		if (this.model.cells.findIndex(c => c.cellUri.toString() === uriString) > -1) {
+			this.selectCell(cell);
+			return cell.runCell(this.notificationService, this.connectionManagementService);
+		} else {
+			return Promise.reject(new Error(localize('cellNotFound', "cell with URI {0} was not found in this model", uriString)));
+		}
+	}
+
+	public async runAllCells(startCell?: ICellModel, endCell?: ICellModel): Promise<boolean> {
+		await this.modelReady;
+		let codeCells = this.model.cells.filter(cell => cell.cellType === CellTypes.Code);
+		if (codeCells && codeCells.length) {
+			// For the run all cells scenario where neither startId not endId are provided, set defaults
+			let startIndex = 0;
+			let endIndex = codeCells.length;
+			if (!isUndefinedOrNull(startCell)) {
+				startIndex = codeCells.findIndex(c => c.id === startCell.id);
+			}
+			if (!isUndefinedOrNull(endCell)) {
+				endIndex = codeCells.findIndex(c => c.id === endCell.id);
+			}
+			for (let i = startIndex; i < endIndex; i++) {
+				let cellStatus = await this.runCell(codeCells[i]);
+				if (!cellStatus) {
+					return Promise.reject(new Error(localize('cellRunFailed', "Run Cells failed - See error in output of the currently selected cell for more information.")));
+				}
+			}
+		}
+		return true;
+	}
+	clearOutput(cell: ICellModel): Promise<boolean> {
+		throw new Error('Method not implemented.');
```

Comment:
Are you planning on implementing these later?

Reply:
Yes, we are still working on this functionality.